### PR TITLE
Update rust dependencies

### PR DESCRIPTION
**Pull request by jpculp**
**Description of changes:**

Ran `cargo update` for `sources` and `tools`, avoided some updates that pull unnecessary extra versions, and adjusted the deny TOMLs as necessary. Also included updates for **AWS SDK for Rust**, `coldsnap`, and `tough`.

**Testing done:**

_Coming soon!_

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by jpculp** [comment]
`coldsnap` and `tough` are temporarily pointing to my forks until their next release. Until then we'll keep this in draft.

**Comment by jpculp** [comment]
Added a temporary commit to fix some hashes and bump some crates.

File: sources/api/migration/migrator/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ regex = "1"
 semver = "1"
 simplelog = "0.12"
 snafu = "0.7"
-tough = "0.13"
+tough = { git = "https://github.com/jpculp/tough", branch="update-rust-deps", version="0.13" }
```
Comment (webern) [inline sources/api/migration/migrator/Cargo.toml:23]: reminder

**Review by webern — CHANGES_REQUESTED** [review]
Looks good other than the temporary github references.

**Comment by jpculp** [comment]
Superseded by #3330.

**Pull request closed without merging**